support-models/src/main/scala/com/gu/support/encoding/CustomCodecs.scala

```diff
@@ -65,7 +66,10 @@ trait EitherCodecs {
   implicit def decodeEither[A, B](implicit
       decoderA: Decoder[A],
       decoderB: Decoder[B],
-  ): Decoder[Either[A, B]] = decoderA.either(decoderB)
+  ): Decoder[Either[A, B]] = {
+    import CodecHelpers.withClue
+    CodecHelpers.or(withClue(decoderA.map(Left.apply), "Left"), withClue(decoderB.map(Right.apply), "Right"))
```

Comment:
when I was debugging the tests I discovered that our `or` function (below) and the circe `either` function just reported the last error message without context, rather than everything that they tried.  This makes for a useless error message.
Although it would be a nice opportunity to use a [discriminator](https://circe.github.io/circe/codecs/adt.html#the-future) that would require extra fields on the client side.  I have just patched it up with a better error message for now.  It's a bit verbose in terms of "here are all the things I tried...." but that can't be helped without the client side tweaks.